pkgs/development/libraries/geogram/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, stdenv
+, cmake
+, fetchFromGitHub
+, python3Packages
+}:
+
+let
+  testdata = fetchFromGitHub {
+    owner = "BrunoLevy";
+    repo = "geogram.data";
+    rev = "70c51a4968960e2d4fa652421b5721672f2b2b70";
+    hash = "sha256-vJtbDDjt2t/6TH0k3i287yGZRd1+qbw20F8D8CFgb3c=";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "geogram";
+  version = "1.8.2";
+
+  src = fetchFromGitHub {
+    owner = "BrunoLevy";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-8VpBPtFMSDr5Al72ckRxHjoM4vOCzMl52MlEYyG+MSs=";
+    fetchSubmodules = true;
+  };
+
+  outputs = [ "bin" "lib" "dev" "out" ];
+
+  cmakeFlags = [
+    # Triangle has unclear licensing situation
+    "-DGEOGRAM_WITH_TRIANGLE=OFF"
+    "-DGEOGRAM_WITH_LEGACY_NUMERICS=OFF"
+    "-DGEOGRAM_WITH_LUA=OFF"
+    "-DGEOGRAM_WITH_GRAPHICS=OFF"
+  ];
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  postPatch = lib.optionalString stdenv.isAarch64 ''
+    substituteInPlace cmake/platforms/*/config.cmake \
+      --replace "-m64" ""
+    '';
+
+  postInstall = ''
+    mkdir -p $lib/lib
+    mv -t $lib/lib $out/lib/*.so*
+
+    mkdir -p $bin/bin
+    mv -t $bin/bin $out/bin/*
+    rmdir $out/bin
+  '';
+
+  # The multiple output hook will remove $out during fixup, so we need to create it again
+  # to prevent Nix from complaining it is missing
```

Comment:
that sound like a general problem. it would be great to find a general solution

could you report it?

Reply:
I will first need to make sure this is not because of my own fault :slightly_smiling_face:
Perhaps I misconfigured something, not sure. Parts of the multiple output machinery seem to operate in.. mysterious ways.

Reply:
I'm sure `$out` is needed for a build to finish successfully. I remember it being the case for package tests.

So removing it seem a strange behavior.

Reply:
resolved in amended commit